source/_static/js/main.js

```diff
@@ -504,4 +504,21 @@ window.addEventListener("DOMContentLoaded", (event) => {
       });
     }
   })();
-});
\ No newline at end of file
+// ------------------------------------------------
+// Tab switching for hidden anchors
+// ------------------------------------------------
+
+    document.getElementById((document.getElementById(window.location.hash.substring(1)))
+        .closest("div")
+        .previousSibling.getAttributes("for"))
```

Comment:
There seems to be an error in this function:

<img width="816" alt="Screenshot 2024-02-21 at 12 30 09" src="https://github.com/minio/docs/assets/13393018/d150e642-b1dd-4015-b686-4792743c41e3">

I think we can add a little more context and rewrite it as shown below:

```
  // Target ID - where the browser need to scroll
  const hashElem = document.getElementById(window.location.hash.substring(1));

  // If the element is hidden, neither scrollIntoView nor hashScroll would work. So this function is to check whether the elem is visible or not.
  function isHashElemeVisible(elem) {
    var style = window.getComputedStyle(elem);
    return (elem.offsetWidth > 0 || elem.offsetHeight > 0) && style.display !== 'none' && style.visibility !== 'hidden';
  }

  // First check if the hash is valid. 
  // And if it's valid but hidden, it means the corresponding tab is hidden. So, make the tab active and  
  if(hashElem) {
    if(!isHashElemeVisible(hashElem)) {
      hashElem.closest(".sd-tab-content").previousElementSibling.click();
    
      setTimeout(() => {
        hashElem.scrollIntoView();
      }, 100);
    }
    else {
      hashElem.scrollIntoView();
    }
  }
});
```